src/address-manager/utils.ts

```diff
@@ -0,0 +1,14 @@
+
+export function debounce (func: () => void, wait: number): () => void {
```

Comment:
This is quite a useful function, I would if we should make it generic for other use cases, such as in DHT queries for example. Something like

```suggestion
function debounce<T extends (...args: any[]) => any>(func: T, delay: number): (...args: Parameters<T>) => void {
  let timeoutId: NodeJS.Timeout;
  return function (...args: Parameters<T>): void {
    clearTimeout(timeoutId);
    timeoutId = setTimeout(() => {
      func(...args);
    }, delay);
  };
}
```

Reply:
I think this might cause hard to debug behaviour because only the args passed to the final call will be used, so lets only make this change when we need it.

For example:

```js
const fn = (arg: string) => {
  console.info('got arg:', arg)
}

const debounced = debounce(fn, 100)

debounced('one')
debounced('two')
debounced('three')
// output:
// got arg: three
```

In this case the caller may not know that the function has been debounced so may expect all three invocations to occur and be left scratching their head when they don't.